proxy-socket-core/src/main/java/net/airvantage/proxysocket/core/v2/ProxyProtocolV2Decoder.java

```diff
@@ -0,0 +1,219 @@
+                throw new ProxyProtocolParseException("Invalid signature");
+            }
+        }
+
+        int pos = offset + PROTOCOL_SIGNATURE.length;
+
+        // Byte 13: version/command
+        int verCmd = data[pos++] & 0xFF;
+        int version = (verCmd >> 4) & 0x0F;
+        if (version != 2) throw new ProxyProtocolParseException("Invalid version");
+        int cmd = verCmd & 0x0F;
+        ProxyHeader.Command command;
+        switch (cmd) {
+            case 0x00:
+                // Early return for LOCAL command
+                return new ProxyHeader(ProxyHeader.Command.LOCAL, ProxyHeader.AddressFamily.AF_UNSPEC, ProxyHeader.TransportProtocol.UNSPEC, null, null, null, PROTOCOL_SIGNATURE_FIXED_LENGTH);
+            case 0x01:
+                command = ProxyHeader.Command.PROXY;
+                break;
+            default:
+                throw new ProxyProtocolParseException("Invalid command");
+        }
+
+        // Byte 14: address family and protocol
+        int famProto = data[pos++] & 0xFF;
+        int fam = famProto & 0xF0;
+        int proto = famProto & 0x0F;
+
+        ProxyHeader.AddressFamily af = parseAddressFamily(fam);
+        ProxyHeader.TransportProtocol tp = parseTransportProtocol(proto);
+
+        // Byte 15, 16: Length of address part of the header, including TLVs
+        int variableLength = ((data[pos++] & 0xFF) << 8) | (data[pos++] & 0xFF);
+
+        // Check if we have enough data for the header
+        int headerLen = PROTOCOL_SIGNATURE_FIXED_LENGTH + variableLength;
+        if (headerLen > length) throw new ProxyProtocolParseException("Insufficient data for header");
+
+        AddressPair addresses = null;
+        if (af != ProxyHeader.AddressFamily.AF_UNSPEC) {
+            addresses = parseAddresses(data, pos, af, variableLength);
+            pos += addresses.bytesConsumed;
+        }
+
+        List<Tlv> tlvs = null;
+        if (parseTlvs) {
+            int tlvLen = Math.max(0, variableLength - (addresses != null ? addresses.bytesConsumed : 0));
+            tlvs = parseTlvs(data, pos, tlvLen);
+        }
+
+        InetSocketAddress src = addresses != null ? addresses.src : null;
+        InetSocketAddress dst = addresses != null ? addresses.dst : null;
+
+        return new ProxyHeader(command, af, tp, src, dst, tlvs, headerLen);
+    }
+
+    private static ProxyHeader.AddressFamily parseAddressFamily(int fam)
+        throws ProxyProtocolParseException {
+        return switch (fam) {
+            case 0x00 -> ProxyHeader.AddressFamily.AF_UNSPEC;
+            case 0x10 -> ProxyHeader.AddressFamily.AF_INET;
+            case 0x20 -> ProxyHeader.AddressFamily.AF_INET6;
+            case 0x30 -> ProxyHeader.AddressFamily.AF_UNIX;
+            default -> throw new ProxyProtocolParseException("Invalid address family");
+        };
+    }
+
+    private static ProxyHeader.TransportProtocol parseTransportProtocol(int proto)
```

Comment:
I'd laos rather have
```
import  net.airvantage.proxysocket.core.v2.ProxyHeader.*;
```
and
```suggestion
    private static TransportProtocol parseTransportProtocol(int proto)
```